sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/internal/state/SynchronousMetricStorageTest.java

```diff
@@ -413,16 +413,15 @@ void recordAndCollect_concurrentStressTest(
               // If we terminate when latch.count() == 0, the last collect may have occurred before
               // the last recorded measurement. To ensure we collect all measurements, we collect
               // one extra time after latch.count() == 0.
-              while (latch.getCount() != 0 && extraCollects <= 1) {
+              while (latch.getCount() != 0 || extraCollects <= 1) {
```

Comment:
It's a little silly to be polling the latch anyway, right? Shouldn't we have a `assertTrue(latch.await(30, SECONDS))` just followed by the extra collect then?

Reply:
Well we do want to do lots of collect at the same time as the records to catch any bugs related to the interaction of collecting and recording. Polling the latch just says ensure we keep collecting on a loop until all the record threads are done.

Reply:
The idea is to do several synchronous collects while the 4 threads do their recordings concurrently. Got it. Seems fine then, that just wasn't instantly obvious on first read.